addons/calendar/static/src/components/activity_delete_confirm_dialog/activity_delete_confirm_dialog.js

```diff
@@ -0,0 +1,87 @@
+/** @odoo-module **/
+
+import useShouldUpdateBasedOnProps from '@mail/component_hooks/use_should_update_based_on_props/use_should_update_based_on_props';
+import useStore from '@mail/component_hooks/use_store/use_store';
+
+import Dialog from 'web.OwlDialog'
+
+const { Component } = owl;
+const { useRef } = owl.hooks;
+
+const components = { Dialog };
+
+class ActivityDeleteConfirmDialog extends Component {
```

Comment:
This seems to add a whole lot of code just for a confirmation dialog.

Is the Dialog.confirm from the old code not "owl compliant"?
I find it very handy to avoid all this template / component definitions.

Reply:
Dialog.confirm was my first idea as well but I couldn't make it work in this case. So, instead, I did the same thing as they did in attachment_delete_confirm_dialog.js found in the mail module. I supposed that if they did it this way it was fine as the objective is the same.

But I do agree with you that it is quite a lot of code for such a small basic thing.

Reply:
I was curious so I played with it a little bit and this seems to work fine for me:

```js
/**
 * @override
 */
_onClickCancel(ev) {
    const superMethod = this._super;
    if (this.activity.category !== 'meeting' && this.activity.category !== 'scheduled_phonecall'){
        this._super(ev);
    } else {
        Dialog.confirm(
            this,
            _t("The activity is linked to a meeting. Deleting it will remove the meeting as well. Do you want to proceed ?"), {
                confirm_callback: function () {
                    superMethod(ev);
                },
            }
        );
    }
}
```

Where I manually imported:

```js
const Dialog = require('web.Dialog');
const core = require('web.core');
const _t = core._t;
```

(And you would keep the override of "deleteServerRecord" of course).

Seems a lot shorter, what do you think?